SDK release for Azure.ResourceManager.Migration.Assessment v2023-03-15

srilipta-swain opened the pull request:
This is a new and first SDK release for Azure.ResourceManager.Migration.Assessment for latest GA version i.e. 2023-03-15.

srilipta-swain committed:
Add samples and tests

srilipta-swain committed:
Merge branch 'main' of https://github.com/Azure/azure-sdk-for-net into srswain/migrateSDK
Merge main branch into srswain/migrateSDK

jsquire reviewed (dismissed):
@srilipta-swain:

Your account lacks the required public membership to the "Microsoft" GitHub organization needed for an internal contributor.  Please review the Azure SDK onboarding documentation and use the associated Teams channel for support.

- [Azure SDK onboarding](https://eng.ms/docs/products/azure-developer-experience/onboard/requirements) _(Microsoft internal)_
- [Azure SDK onboarding assistance](https://teams.microsoft.com/l/channel/19%3ade3fce22c2994be18cac50502c55f717%40thread.skype/Azure%2520SDK%2520Onboarding?groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47) _(Microsoft internal)_

You can verify the state of your account by running the [Validate-AzsdkCodeOwner](https://github.com/Azure/azure-sdk-tools/blob/main/tools/github/scripts/Validate-AzsdkCodeOwner.ps1) script from the Azure SDK tools repository.

Please also be sure to add yourself to [CODEOWNERS](https://github.com/Azure/azure-sdk-for-net/blob/main/.github/CODEOWNERS) for this library, if you will be maintaining it going forward.

srilipta-swain committed:
Added more tests

srilipta-swain commented:
> @srilipta-swain:
> 
> Your account lacks the required public membership to the "Microsoft" GitHub organization needed for an internal contributor. Please review the Azure SDK onboarding documentation and use the associated Teams channel for support.
> 
> * [Azure SDK onboarding](https://eng.ms/docs/products/azure-developer-experience/onboard/requirements) _(Microsoft internal)_
> * [Azure SDK onboarding assistance](https://teams.microsoft.com/l/channel/19%3ade3fce22c2994be18cac50502c55f717%40thread.skype/Azure%2520SDK%2520Onboarding?groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47) _(Microsoft internal)_
> 
> You can verify the state of your account by running the [Validate-AzsdkCodeOwner](https://github.com/Azure/azure-sdk-tools/blob/main/tools/github/scripts/Validate-AzsdkCodeOwner.ps1) script from the Azure SDK tools repository.
> 
> Please also be sure to add yourself to [CODEOWNERS](https://github.com/Azure/azure-sdk-for-net/blob/main/.github/CODEOWNERS) for this library, if you will be maintaining it going forward.

Addressed the points. Thanks for your review Jesse.

srilipta-swain committed:
Merge main

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[Azure.ResourceManager.Migration.Assessment](https://apiview.dev/Assemblies/Review/b3366da54a6d485da0e844eed744f6b2?revisionId=27f684ef216142338d9f9fbb41966404)

live1206 commented on sdk/migrationassessment/Azure.ResourceManager.Migration.Assessment/src/autorest.md:
```diff
@@ -0,0 +1,261 @@
+  GroupUpdateOperation: MigrateGroupUpdateOperationType
+  MachineBootType.EFI: Efi
+  AssessmentProjectUpdate.properties.customerWorkspaceId: -|arm-id
+  AzureOfferCode.MSAZR0003P: MSAZR0003P
```
The names of L55 to L93 remain the same, what are these configurations for?

srilipta-swain replied:
These are the OfferCodes used during Assessment evaluation done by the service

live1206 commented on sdk/migrationassessment/Azure.ResourceManager.Migration.Assessment/assets.json:
```diff
@@ -0,0 +1,6 @@
+  "AssetsRepo": "Azure/azure-sdk-assets",
+  "AssetsRepoPrefixPath": "net",
+  "TagPrefix": "net/migration.assessment/Azure.ResourceManager.Migration.Assessment",
+  "Tag": ""
```
Please upload the test recordings to let the tests to run in playback mode.
Details can be found in https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/resourcemanager/Azure.ResourceManager/docs/TestGuide.md

srilipta-swain replied:
Thanks @live1206 , added them. Here is the [link](https://github.com/Azure/azure-sdk-assets/releases/tag/net%2Fmigration.assessment%2FAzure.ResourceManager.Migration.Assessment_1035c2eaca).

srilipta-swain committed:
Updated test recordings Tag in assets.json

ArthurMa1978 commented:
/azp run prepare-pipelines

ArthurMa1978 committed:
Merge branch 'main' into migrate

ArthurMa1978 committed:
Update

ArthurMa1978 committed:
Update

ArthurMa1978 committed:
update

ArthurMa1978 committed:
Merge branch 'main' into migrate

ArthurMa1978 committed:
Update

ArthurMa1978 committed:
Update

srilipta-swain committed:
Updated the tests

ArthurMa1978 committed:
Merge branch 'main' into migrate

ArthurMa1978 committed:
Merge branch 'main' into migrate

ArthurMa1978 committed:
update

ArthurMa1978 commented on sdk/migrationassessment/Azure.ResourceManager.Migration.Assessment/README.md:
```diff
@@ -0,0 +1,80 @@
+# Microsoft Azure Migration.Assessment management client library for .NET
+
+**[Describe the service briefly first.]**
```
Please add description of your servicer here.

scbedd commented:
/azp run net - pullrequest

scbedd commented:
/azp run net - resourcemanager - mgmt - ci

ArthurMa1978 committed:
update

ArthurMa1978 committed:
Merge branch 'main' into migrate

srilipta-swain committed:
Updated Readme

srilipta-swain committed:
Merge branch 'srswain/migrateSDK' of https://github.com/Azure/azure-sdk-for-net into srswain/migrateSDK

ArthurMa1978 committed:
Merge branch 'srswain/migrateSDK' of https://github.com/Azure/azure-sdk-for-net into migrate

ArthurMa1978 committed:
update

ArthurMa1978 committed:
Merge branch 'main' into migrate

ArthurMa1978 committed:
update
